### PR TITLE
add news feature

### PR DESCRIPTION
**Pull request by badmon**
news functionality implemented using new york times articles api

### CODE REVIEW

**Comment by badmon** [comment]
wit ai needs training to get intent of query in test_news

File: modules/src/news.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (swapagarwal) [inline modules/src/news.py]: empty lines

**Comment by swapagarwal** [comment]
Hey @badmon, nice work!
By the way, did you see the attribution guidelines of the API?

**Comment by badmon** [comment]
I just did, seems to me we aren't breaking any rules ?

**Comment by badmon** [comment]
what further progress is there before  branch gets merged?

**Comment by swapagarwal** [comment]
See https://developer.nytimes.com/branding, need to add "Data provided by The New York Times" at the end

**Comment by swapagarwal** [comment]
I've trained wit.ai, the tests are now passing! :tada: 
Also, can you rebase your work on master so that merge conflicts are resolved?

**Comment by badmon** [comment]
deleted my branch by mistaken -- is it possible for me to create another news-feature branch ?

**Comment by badmon** [comment]
oh my bad -- git reflog :+1:

**Comment by badmon** [comment]
funny thing is cant seem to find a template that allows me to display "data provided by new york times"

**Comment by swapagarwal** [comment]
Keep the snippet in the subtitle, and put the attribution in place of 'View on Web'.

**Comment by badmon** [comment]
alright, on it!

**Comment by swapagarwal** [comment]
still merge conflicts, kindly sync your fork with upstream

**Comment by badmon** [comment]
i have, still having conflicts

**Comment by swapagarwal** [comment]
The conflict is in line number 6 in config.py
JOKES_SOURCE_FILE = 'data/jokes.json'

**Comment by badmon** [comment]
done ?

File: modules/src/anime.py [hunk 2]
```diff
@@ -12,7 +12,9 @@ def process(input, entities):
 
         template = TextTemplate()
         template.set_text('Title: ' + data[0]['title'] + '\nSynopsis: ' + data[0]['synopsis'])
-        template.set_post_text('\nCommunity Rating: ' + str(data[0]['community_rating']) + '\nStatus: ' + data[0]['status'])
+        #only want to see 2 decimal places
+        rating = {0:.2f}.format(data[0]['community_rating'])
+        template.set_post_text('\nCommunity Rating: ' + str(rating) + '\nStatus: ' + data[0]['status'])
```
Comment (swapagarwal) [inline modules/src/anime.py]: this goes in separate PR

**Comment by badmon** [comment]
i just realized how terrible i am at git, i will do that soon

**Comment by swapagarwal** [comment]
No problem, take your time :smile: Let me know if you need any help!
Using it is how you learn. Cheers!

**Comment by badmon** [comment]
want commands do i run to resolve the conflicts ? i tried git fetch upstream && git merge upstream/master

**Comment by swapagarwal** [comment]
Nice work! Conflicts have been resolved.
Just one last thing, there should be changes in only 4 files [here](https://github.com/swapagarwal/JARVIS-on-Messenger/pull/48/files)

File: data/jokes.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
+
```
Comment (swapagarwal) [inline data/jokes.json]: no need

**Comment by swapagarwal** [comment]
Merged in #147

**Pull request closed without merging**